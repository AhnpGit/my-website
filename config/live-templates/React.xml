<templateSet group="React">
  <template name="rx_module" value="const _ACTION_PREFIX = '$projectName$/$moduleName$'&#10;&#10;const $ACTION_NAME$ = `${_ACTION_PREFIX}/$ACTION_NAME$`&#10;&#10;// TODO: Connect the reducer.&#10;export default function reducer(state = {$defaultState$}, action) {&#10;    switch (action.type) {&#10;        case $ACTION_NAME$:&#10;            return {$END$}&#10;&#10;        default:&#10;            return state&#10;    }&#10;}&#10;&#10;export const $actionCreator$ = () =&gt; ({&#10;    type: $ACTION_NAME$,&#10;    payload: {$payload$},&#10;    meta: {},&#10;    error: undefined,&#10;})&#10;" description="Creates a DUCKS-module" toReformat="true" toShortenFQNames="true">
    <variable name="projectName" expression="" defaultValue="&quot;&lt;project name&gt;&quot;" alwaysStopAt="true" />
    <variable name="moduleName" expression="fileNameWithoutExtension()" defaultValue="" alwaysStopAt="true" />
    <variable name="ACTION_NAME" expression="" defaultValue="&quot;MY_ACTION&quot;" alwaysStopAt="true" />
    <variable name="defaultState" expression="" defaultValue="" alwaysStopAt="true" />
    <variable name="actionCreator" expression="" defaultValue="&quot;myActionCreator&quot;" alwaysStopAt="true" />
    <variable name="payload" expression="" defaultValue="" alwaysStopAt="true" />
    <context>
      <option name="JS_STATEMENT" value="true" />
    </context>
  </template>
  <template name="rx_component" value="import React from 'react'&#10;&#10;const $ComponentName$ = ({$props$}) =&gt; (&#10;    &lt;$END$/&gt;&#10;)&#10;&#10;export default $ComponentName$" description="Creates a functional component" toReformat="true" toShortenFQNames="true">
    <variable name="ComponentName" expression="capitalize(camelCase(fileNameWithoutExtension()))" defaultValue="" alwaysStopAt="true" />
    <variable name="props" expression="" defaultValue="" alwaysStopAt="true" />
    <context>
      <option name="JS_STATEMENT" value="true" />
    </context>
  </template>
  <template name="rx_component_class" value="import React from 'react'&#10;&#10;class $ComponentName$ extends React.Component {&#10;    render() {&#10;        return (&#10;            &lt;$END$/&gt;&#10;        )&#10;    }&#10;}&#10;&#10;export default $ComponentName$" description="Creates a class component" toReformat="true" toShortenFQNames="true">
    <variable name="ComponentName" expression="capitalize(camelCase(fileNameWithoutExtension()))" defaultValue="" alwaysStopAt="true" />
    <context>
      <option name="JS_STATEMENT" value="true" />
    </context>
  </template>
  <template name="rx_proptypes" value="import PropTypes from 'prop-types'&#10;$ComponentName$.propTypes = {&#10;    $prop$: PropTypes.$END$,&#10;}" description="Adds proptypes to a component" toReformat="true" toShortenFQNames="true">
    <variable name="ComponentName" expression="capitalize(camelCase(fileNameWithoutExtension()))" defaultValue="" alwaysStopAt="true" />
    <variable name="prop" expression="" defaultValue="&quot;prop&quot;" alwaysStopAt="true" />
    <context>
      <option name="JS_STATEMENT" value="true" />
    </context>
  </template>
  <template name="rx_action_async" value="const $ACTION_NAME$ = `${_ACTION_PREFIX}/$ACTION_NAME$`&#10;const $ACTION_NAME$_PENDING = `${$ACTION_NAME$}_PENDING`&#10;const $ACTION_NAME$_FULFILLED = `${$ACTION_NAME$}_FULFILLED`&#10;const $ACTION_NAME$_REJECTED = `${$ACTION_NAME$}_REJECTED`" description="Creates async action types" toReformat="true" toShortenFQNames="true">
    <variable name="ACTION_NAME" expression="" defaultValue="&quot;MY_ACTION&quot;" alwaysStopAt="true" />
    <context>
      <option name="JS_STATEMENT" value="true" />
    </context>
  </template>
  <template name="rx_action" value="const $ACTION_NAME$ = `${_ACTION_PREFIX}/$ACTION_NAME$`" description="Creates an action type" toReformat="true" toShortenFQNames="true">
    <variable name="ACTION_NAME" expression="" defaultValue="&quot;MY_ACTION&quot;" alwaysStopAt="true" />
    <context>
      <option name="JS_STATEMENT" value="true" />
    </context>
  </template>
  <template name="rx_creator" value="export const $actionCreator$ = () =&gt; ({&#10;    type: $ACTION_NAME$,&#10;    payload: {$END$},&#10;    meta: {},&#10;    error: undefined,&#10;})" description="Creates an action creator" toReformat="true" toShortenFQNames="true">
    <variable name="actionCreator" expression="" defaultValue="&quot;myActionCreator&quot;" alwaysStopAt="true" />
    <variable name="ACTION_NAME" expression="" defaultValue="&quot;MY_ACTION&quot;" alwaysStopAt="true" />
    <context>
      <option name="JS_STATEMENT" value="true" />
    </context>
  </template>
  <template name="rx_creator_async" value="export const $actionCreator$ = () =&gt; ({&#10;    type: $ACTION_NAME$,&#10;    payload: getPayload(),&#10;    meta: {},&#10;    error: undefined,&#10;    &#10;    async function getPayload() {&#10;        return $END$&#10;    }&#10;})" description="Creates an async action creator" toReformat="true" toShortenFQNames="true">
    <variable name="actionCreator" expression="" defaultValue="&quot;myActionCreator&quot;" alwaysStopAt="true" />
    <variable name="ACTION_NAME" expression="" defaultValue="&quot;MY_ACTION&quot;" alwaysStopAt="true" />
    <context>
      <option name="JS_STATEMENT" value="true" />
    </context>
  </template>
  <template name="rx_container" value="import { connect } from 'react-redux'&#10;import $ComponentName$ from '../components/$ComponentName$'&#10;&#10;const mapStateToProps = (state) =&gt; ({&#10;    $END$&#10;})&#10;&#10;const mapDispatchToProps = (dispatch) =&gt; ({&#10;})&#10;&#10;export default connect(&#10;    mapStateToProps,&#10;    mapDispatchToProps,&#10;)($ComponentName$)" description="Creates a container class" toReformat="true" toShortenFQNames="true">
    <variable name="ComponentName" expression="capitalize(camelCase(fileNameWithoutExtension()))" defaultValue="" alwaysStopAt="true" />
    <context>
      <option name="JS_STATEMENT" value="true" />
    </context>
  </template>
  <template name="rx_styled" value="import styled from 'styled-components'&#10;&#10;const $ComponentName$ = styled.$ElementType$`&#10;  $END$&#10;`&#10;&#10;export default $ComponentName$" description="Creates a styled component" toReformat="true" toShortenFQNames="true">
    <variable name="ComponentName" expression="capitalize(camelCase(fileNameWithoutExtension()))" defaultValue="" alwaysStopAt="true" />
    <variable name="ElementType" expression="" defaultValue="&quot;div&quot;" alwaysStopAt="true" />
    <context>
      <option name="JS_STATEMENT" value="true" />
    </context>
  </template>
</templateSet>
