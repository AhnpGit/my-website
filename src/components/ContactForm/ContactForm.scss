.wrapper__contact {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  background-color: $white;

  @include media-query($medium-up) {
    padding: 0 $base-unit;
  }

  .contact__intro {
    h2 {
      text-align: center;
      word-spacing: 2px;
      letter-spacing: 1px;
      text-transform: uppercase;
      margin-bottom: $base-unit;

      &:after {
        display: block;
        height: 2px;
        background-color: $tahitiGold;
        content: " ";
        width: 100px;
        margin: 0 auto;
        margin-top: $base-unit;
      }
    }
  }

  .contact__body {
    width: 90%;

    @include media-query($medium-up) {
      width: 60%;
    }

    @include media-query($large-up) {
      width: 50%;
    }

    .contact__form {
      input[type=text],
      input[type=email],
      select,
      textarea {
        padding: $base-spacer;
        border-radius: 3px;
        border: 1px solid #ccc;
      }

      textarea {
        height: 100px;
      }

      input[type=checkbox] {
        margin: $base-spacer 5px $base-spacer 0;
      }

      .form__row {
        lost-center: 100% 0 flex;
        align-items: center;
        margin-bottom: $base-unit;

        label {
          lost-row: 1/3 0;
          margin-bottom: $base-spacer;
        }

        input[type=text],
        input[type=email],
        select,
        textarea {
          flex: 1 0 auto;
        }

        @include media-query($medium-up) {
          label {
            lost-column: 1/3 0 0;
            margin-bottom: 0;
          }

          input[type=text],
          input[type=email],
          select,
          textarea {
            lost-column: 2/3 0 0;
          }
        }
      }

      .form__row:not(:last-child) {
        @include step-scale('font-size', 85%, 100%, 100%);
      }

      .form__row:last-child {
        justify-content: center;
      }
    }
  }
}
