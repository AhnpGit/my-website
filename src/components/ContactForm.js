import React from 'react'

const ContactForm = () => (
  <section className="section-form">
    <div className="row">
      <h2>We're happy to hear from you</h2>
    </div>
    <div className="row">
      <form method="post" action="#" className="contact-form">
        <div className="row">
          <div className="col span-1-of-3">
            <label htmlFor="name">Name</label>
          </div>
          <div className="col span-2-of-3">
            <input type="text" name="name" id="name" placeholder="Your name" required />
          </div>
        </div>
        <div className="row">
          <div className="col span-1-of-3">
            <label htmlFor="email">Email</label>
          </div>
          <div className="col span-2-of-3">
            <input type="email" name="email" id="email" placeholder="Your email" required />
          </div>
        </div>
        <div className="row">
          <div className="col span-1-of-3">
            <label htmlFor="find-us">How did you find us?</label>
          </div>
          <div className="col span-2-of-3">
            <select name="find-us" id="find-us">
              <option value="friends">Friends</option>
              <option value="search">Search engine</option>
              <option value="ad">Advertisement</option>
              <option value="other">Other</option>
            </select>
          </div>
        </div>
        <div className="row">
          <div className="col span-1-of-3">
            <label htmlFor="news">Newsletter?</label>
          </div>
          <div className="col span-2-of-3">
            <input type="checkbox" name="news" id="news" /> Yes, please
          </div>
        </div>
        <div className="row">
          <div className="col span-1-of-3">
            <label htmlFor="message">Drop us a line</label>
          </div>
          <div className="col span-2-of-3">
            <textarea name="message" placeholder="Your message"></textarea>
          </div>
        </div>
        <div className="row">
          <div className="col span-1-of-3">
            <label htmlFor="submit">&nbsp;</label>
          </div>
          <div className="col span-2-of-3">
            <input name="submit" type="submit" value="Send it!" />
          </div>
        </div>
      </form>
    </div>
  </section>
)

export default ContactForm
