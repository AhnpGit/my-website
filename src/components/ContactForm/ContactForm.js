// @flow

import React from 'react'
import Wrapper from '../../elements/Wrapper/Wrapper'
import Row from '../../elements/Row/Row'
import Button from '../../elements/Button/Button'

import './ContactForm.scss'

/**
 * Contact form component displays a contact form
 */
const ContactForm = () => (
  <Row className="contact">
    <Wrapper name="contact">
      <div className="contact__intro">
        <h2>We're happy to hear from you</h2>
      </div>
      <div className="contact__body">
        <form method="post" action="#" className="contact__form">
          <div className="form__row">
            <label htmlFor="name">Name</label>
            <input type="text" name="name" className="name" placeholder="Your name" required />
          </div>
          <div className="form__row">
            <label htmlFor="email">Email</label>
            <input type="email" name="email" className="email" placeholder="Your email" required />
          </div>
          <div className="form__row">
            <label htmlFor="find-us">How did you find us?</label>
            <select name="find-us" id="find-us">
              <option value="friends">Friends</option>
              <option value="search">Search engine</option>
              <option value="ad">Advertisement</option>
              <option value="other">Other</option>
            </select>
          </div>
          <div className="form__row">
            <label htmlFor="news">Newsletter?</label>
            <input type="checkbox" name="news" className="news" />Yes, please!
          </div>
          <div className="form__row">
            <label htmlFor="message">Drop us a line</label>
            <textarea name="message" placeholder="Your message" defaultValue="" />
          </div>
          <div className="form__row">
            <Button text="Send it!" onClick={() => {}}/>
          </div>
        </form>
      </div>
    </Wrapper>
  </Row>
)

export default ContactForm
