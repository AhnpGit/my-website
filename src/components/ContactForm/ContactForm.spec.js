import React from 'react'
import { shallow } from 'enzyme'
import ContactForm from './ContactForm'

describe('ContactForm component', () => {
  it('renders correctly without any issues', () => {
    const result = shallow(
      <ContactForm />
    )

    expect(result).toMatchSnapshot()
  })
})
