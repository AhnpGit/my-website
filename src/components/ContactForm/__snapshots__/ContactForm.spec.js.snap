// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactForm component renders correctly without any issues 1`] = `
<Row
  className="contact"
>
  <Wrapper
    name="contact"
  >
    <div
      className="contact__intro"
    >
      <h2>
        We're happy to hear from you
      </h2>
    </div>
    <div
      className="contact__body"
    >
      <form
        action="#"
        className="contact__form"
        method="post"
      >
        <div
          className="form__row"
        >
          <label
            htmlFor="name"
          >
            Name
          </label>
          <input
            className="name"
            name="name"
            placeholder="Your name"
            required={true}
            type="text"
          />
        </div>
        <div
          className="form__row"
        >
          <label
            htmlFor="email"
          >
            Email
          </label>
          <input
            className="email"
            name="email"
            placeholder="Your email"
            required={true}
            type="email"
          />
        </div>
        <div
          className="form__row"
        >
          <label
            htmlFor="find-us"
          >
            How did you find us?
          </label>
          <select
            id="find-us"
            name="find-us"
          >
            <option
              value="friends"
            >
              Friends
            </option>
            <option
              value="search"
            >
              Search engine
            </option>
            <option
              value="ad"
            >
              Advertisement
            </option>
            <option
              value="other"
            >
              Other
            </option>
          </select>
        </div>
        <div
          className="form__row"
        >
          <label
            htmlFor="news"
          >
            Newsletter?
          </label>
          <input
            className="news"
            name="news"
            type="checkbox"
          />
          Yes, please!
        </div>
        <div
          className="form__row"
        >
          <label
            htmlFor="message"
          >
            Drop us a line
          </label>
          <textarea
            defaultValue=""
            name="message"
            placeholder="Your message"
          />
        </div>
        <div
          className="form__row"
        >
          <Button
            onClick={[Function]}
            text="Send it!"
          />
        </div>
      </form>
    </div>
  </Wrapper>
</Row>
`;
